### ISSUE TITLE
Code-splitting adds dynamic imports in the same bundle as `react-dom`

### ISSUE THREAD

**Issue by Toolo** [gatsbyjs/gatsby#20044]
## Description

When using `React.lazy()` (loading client-side only code), dependencies are included inside `commons.js` along `react-dom` instead of on a different module.

`react-dom` is needed to hydrate the app, but other client bundles may not be needed until a later stage (ie, code that's only loaded after clicking a button, or interaction side effects)

I've been able to achieve the expected behavior by using `setWebpackConfig` and renaming the `react` cache group to `react` (Gatbsy source to override can be found [here](https://github.com/gatsbyjs/gatsby/blob/823988dc8aa714f09d92b678d5cd373be2fa8a71/packages/gatsby/src/utils/webpack.config.js#L491))

I can create a PR to update it if it makes sense as a common use case

### Steps to reproduce

1. Lazy load a library using `React.lazy()`

```typescript
const MyLazyComponent = React.lazy(() => import('some-large-lib'));

export function MyComponent() {
  // Simple demo check, we can use a `useEffect` hook along a `useState` to tell when we're in client-side.
   if (SERVER) {
      return null;
   }

   return (
      <Suspense fallback={<></>}>
       <MyLazyComponent />
     </Suspense>
   );
}

```

2. Check build results and verify there's a single `commons.js` including both `react-dom` and `some-large-lib`.

### Expected result

What should happen?

- `react-dom` should be included in its own module as it's not likely to change and it's going to be required before lazy modules.
- `some-large-lib` should exist in a different module.

### Actual result

Both `react-dom` and my `some-large-lib` are included in a large `commons.js` file.

### Environment

```
  System:
    OS: macOS 10.15.1
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
    Yarn: 1.19.1 - ~/.yarn/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 78.0.3904.108
    Firefox: 57.0.4
    Safari: 13.0.3
```

**Comment by wardpeet** [comment]
@Toolo I've looked at our .org site where I did some updates to our wepback config but the speedup is hardly noticable so I'm happy to checkout your site to see what effect it has for you, if you can share a repo.

**Comment by Toolo** [comment]
We haven't shipped this to the public yet, but I have some screenshots from `webpack-visualizer` before and after:

Before. Using `react.lazy` and allowing the default configuration to build the bundles.
<img width="1675" alt="before" src="https://user-images.githubusercontent.com/1233927/71006786-d1ec3180-20dd-11ea-91f3-3a45572af316.png">

After. Custom configuration to move `react-dom` into its own bundle.
<img width="1676" alt="after" src="https://user-images.githubusercontent.com/1233927/71006796-d44e8b80-20dd-11ea-8a51-1af850b1d6e4.png">

**Comment by wardpeet** [comment]
What are you gaining by your new config? It still load client.js in your commons file. What would your ideal situation be?

**Comment by Toolo** [comment]
Sorry I didn't come back! Hope you had some great holidays and a good start for 2020 😃 

Let me rephrase my goal:
1. Current scenario: `client.js` is loaded alongside critical JS (react-dom)
2. Intended scenario: `client.js` can be deferred until it's required (app is interactive)

By implementing 2, the app is hydrated earlier on and we reduce time to interactive

**Comment by wardpeet** [comment]
We're marking this issue as answered and closing it for now but please feel free to reopen this and comment if you would like to continue this discussion. We hope we managed to help and thank you for using Gatsby! 💜
